semgrep/semgrep/rule_match.py

```diff
@@ -246,6 +256,30 @@ def get_syntactic_id(self) -> str:
         hash_bytes = int.to_bytes(hash_int, byteorder="big", length=16, signed=False)
         return str(binascii.hexlify(hash_bytes), "ascii")
 
+    # This will supercede syntactic id, as currently that will change even if
+    # things formatting + line numbers change. By using the formula +
+    # metavariable content itself, we remain sensitive to modifications to a
+    # match, but we no longer count formatting + line number changs + other
+    # things as new findings
+    @match_based_id.default
+    def get_match_based_id(self) -> str:
+        try:
+            path = self.path.relative_to(Path.cwd())
+        except ValueError:
+            path = self.path
+        match_formula_str = self.match_formula_string
+        if self.extra.get("metavars") is not None:
+            metavars = self.extra["metavars"]
+            for metavar in metavars:
+                match_formula_str = match_formula_str.replace(
+                    metavar, metavars[metavar]["abstract_content"]
+                )
+        match_id = (match_formula_str, path, self.rule_id)
+        match_id_str = str(match_id)
+        return (
+            f"{hashlib.blake2b(str.encode(match_id_str)).hexdigest()}_{str(self.index)}"
```

Comment:
Is there a reason we're picking this hash function?

Reply:
Yes, https://www.blake2.net/ it's faster than md5, sha-1,2,3 AND cryptographic!

Reply:
It's used by openssl, wireguard, and the linux kernel too